### PR TITLE
Update CA1068 to handle api_surface option

### LINKED ISSUE #2851: CA1068 (CancellationTokenParametersMustComeLast) should be configurable to allow not flagging public APIs
Analyzer should ideally respect [api_surface](https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#analyzed-api-surface) editorconfig option for end user configuration.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #2851

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs [hunk 1]
```diff
@@ -63,6 +63,11 @@ public override void Initialize(AnalysisContext context)
                         return;
                     }
 
+                    if (!symbolContext.Options.MatchesConfiguredVisibility(Rule, methodSymbol, symbolContext.Compilation, symbolContext.CancellationToken))
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs:66]: I think we should not alter default semantics of this analyzer - it should flag all APIs by default, but end users can configure it to only configure specific API surfaces.
```suggestion
                    if (!symbolContext.Options.MatchesConfiguredVisibility(Rule, methodSymbol, symbolContext.Compilation, symbolContext.CancellationToken, defaultRequiredVisibility: SymbolVisibilityGroup.All))
```

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs [hunk 2]
```diff
@@ -27,13 +27,13 @@ public async Task DiagnosticForMethod()
         {
             var source = @"
 using System.Threading;
-class T
+public class T
```
Comment (mavasani) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs:30]: The changes to existing tests should be reverted.

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs [hunk 3]
```diff
@@ -418,5 +418,96 @@ public class C
     }
 }");
         }
+
+        [Theory, WorkItem(2851, "https://github.com/dotnet/roslyn-analyzers/issues/2851")]
+        // General analyzer option
+        [InlineData("public", "dotnet_code_quality.api_surface = public")]
```
Comment (mavasani) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs:424]: Can we add a scenarios to the new tests with empty editorconfig and verify it flags all accessibility APIs?

**Review by mavasani — CHANGES_REQUESTED** [review]
Lets ensure default behavior of the analyzer does not change.

**Comment by Evangelink** [comment]
I totally missed this parameter! I will update the PR now

**Review by mavasani — APPROVED** [review]
Thanks @Evangelink!

**Comment by mavasani** [comment]
@Evangelink Can you please also file a doc issue OR create a doc PR to mention configurability of API surface for https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1068?

**Comment by mavasani** [comment]
Ah, seems the section is already there and points back to the issue your PR fixes: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1068#configurability

**Comment by Evangelink** [comment]
Still needs to be updated, I will file a PR tonight.

**Comment by Evangelink** [comment]
https://github.com/dotnet/docs/pull/21853
